test/e2e/provisioning/testdata/machinedeployment-aws.yaml

```diff
@@ -41,7 +41,7 @@ spec:
               "KubernetesCluster": "randomString"
             # Disabling the public IP assignment requires a private subnet with internet access.
             assignPublicIP: true
-          # Can be 'ubuntu', 'coreos' or 'centos'
+          # Can be 'ubuntu', 'coreos' , 'centos' or 'sles'
```

Comment:
That's correct. Plus, the requirement for this feature was only to be deployed on one cloud provider. Another cloud provider that supports cloud-init was Alibaba but since this provider is kinda experimental I didn't include it.

Reply:
No problem, but then please make sure that all other provides reject it in their validation

Reply:
@moadqassem did you verify that all providers other than AWS reject sles in their validation? Other than that and the mergeconflict, PR lgtm

Reply:
All the providers would reject the SLES except Kubevirt( Added a code to reject sles), Openstack and Vsphere don't reject any OS AFAIK, hypothetically if the images which are used with those providers have cloud-init support they should work. I have excluded them explicitly unless if you have another opinion.